File: ujbm-back/src/main/java/pe/edu/bausate/app/infraestructure/config/ApiPaths.java
package pe.edu.bausate.app.infraestructure.config;

public class ApiPaths {
    private static final String VERSION = "v1";

    public static final String API_BASE = "/api/" + VERSION;
    public static final String ACTUATOR = "/actuator";
    public static final String DOCUMENTATION = "/ujbm-docs";
    public static final String SWAGGER_UI = "/swagger-ui";
    public static final String SWAGGER_RESOURCES = "/v3/api-docs";
    public static final String AUTH = API_BASE + "/auth";
    public static final String REPORT = API_BASE + "/report";
    public static final String PUBLIC_LEVEL = API_BASE + "/public";
    public static final String ADMIN_LEVEL =  API_BASE + "/admin";
    public static final String STUDENT_LEVEL = API_BASE + "/student";
    public static final String TEACHER_LEVEL = API_BASE + "/teacher";
    public static final String INTERVIEWER_LEVEL = API_BASE + "/interviewer";

    // Public endpoints
    public static final String ENUMS_PUBLIC = PUBLIC_LEVEL +"/enums";
    public static final String PROGRAM_PUBLIC = PUBLIC_LEVEL + "/program";
    public static final String APPLICANT_PUBLIC = PUBLIC_LEVEL + "/applicant";
    public static final String UBIGEO_PUBLIC =  PUBLIC_LEVEL +"/ubigeo";
    public static final String GOVERNMENT_PUBLIC = PUBLIC_LEVEL + "/government";

    // Admin endpoints
    public static final String APPLICANT_ADMIN = ADMIN_LEVEL + "/applicant";
    public static final String COURSE_SECTION_ADMIN = ADMIN_LEVEL + "/course-section";
    public static final String COURSE_ADMIN = ADMIN_LEVEL + "/course";
    public static final String PARAMETER_ADMIN = ADMIN_LEVEL + "/system-parameter";
    public static final String TEACHER_ADMIN = ADMIN_LEVEL + "/teacher";
    public static final String STUDENT_ADMIN = ADMIN_LEVEL + "/student";
    public static final String ACADEMIC_PERIOD_ADMIN = ADMIN_LEVEL + "/academic-period";
    public static final String ENROLLMENT_ADMIN = ADMIN_LEVEL + "/enrollment";
    public static final String DASHBOARD_ADMIN = ADMIN_LEVEL + "/dashboard";
    public static final String EXAM_ADMIN = ADMIN_LEVEL + "/exam";
    public static final String INTERVIEWER_ADMIN = ADMIN_LEVEL + "/interviewer";
    public static final String APPOINTMENT_ADMIN = ADMIN_LEVEL + "/appointment";
    // Student endpoints
    public static final String COURSE_SECTION_STUDENT = STUDENT_LEVEL + "/course-section";
    public static final String ENROLLMENT_STUDENT = STUDENT_LEVEL + "/enrollment";

    // Teacher endpoints
    public static final String COURSE_SECTION_TEACHER = TEACHER_LEVEL + "/course-section";

    //interviewer endpoints

}

File: ujbm-front/src/app/shared/utils/api-paths.ts
import { environment } from "@environments/environment";

export class ApiPaths {
   private static readonly HOST = environment.backendUrl;
  private static readonly VERSION = environment.backendVersion;
  private static readonly API_PREFIX = environment.apiPrefix;
  
  // Base paths
  public static readonly API_BASE = `${ApiPaths.API_PREFIX}/${ApiPaths.VERSION}`;
  public static readonly ROOT = ApiPaths.API_BASE;
  public static readonly AUTH = `${ApiPaths.API_BASE}/auth`;
  public static readonly PUBLIC_LEVEL = `${ApiPaths.API_BASE}/public`;
  public static readonly ADMIN_LEVEL = `${ApiPaths.API_BASE}/admin`;
  public static readonly TEACHER_LEVEL = `${ApiPaths.API_BASE}/teacher`;
  public static readonly STUDENT_LEVEL = `${ApiPaths.API_BASE}/student`;
   
  //ENUM
  public static readonly ENUMS = `${ApiPaths.PUBLIC_LEVEL}/enums`;
  public static readonly ENUM_ENROLLLMENT_STATUS_PUBLIC = `${ApiPaths.ENUMS}/enrollment-status`;
  public static readonly ENUM_ENROLLMENT_DETAIL_STATUS_PUBLIC = `${ApiPaths.ENUMS}/enrollment-detail-status`;
  public static readonly ENUM_VALUE_TYPE_PUBLIC = `${ApiPaths.ENUMS}/value-type`;

  // Public endpoints
  public static readonly ENUMS_PUBLIC = `${ApiPaths.PUBLIC_LEVEL}/enums`;
  public static readonly PROGRAM_PUBLIC = `${ApiPaths.PUBLIC_LEVEL}/program`;
  public static readonly APPLICANT_PUBLIC = `${ApiPaths.PUBLIC_LEVEL}/applicant`;
  public static readonly UBIGEO_PUBLIC = `${ApiPaths.PUBLIC_LEVEL}/ubigeo`;
  public static readonly GOVERNMENT_PUBLIC = `${ApiPaths.PUBLIC_LEVEL}/government`;
 
  // Admin endpoints
  public static readonly APPLICANT_ADMIN = `${ApiPaths.ADMIN_LEVEL}/applicant`;
  public static readonly COURSE_SECTION_ADMIN = `${ApiPaths.ADMIN_LEVEL}/course-section`;
  public static readonly COURSE_ADMIN = `${ApiPaths.ADMIN_LEVEL}/course`;
  public static readonly STUDENT_ADMIN = `${ApiPaths.ADMIN_LEVEL}/student`;
  public static readonly PARAMETER_ADMIN = `${ApiPaths.ADMIN_LEVEL}/system-parameter`;
  public static readonly TEACHER_ADMIN = `${ApiPaths.ADMIN_LEVEL}/teacher`;
  public static readonly ACADEMIC_PERIOD_ADMIN = `${ApiPaths.ADMIN_LEVEL}/academic-period`;
  public static readonly ENROLLMENT_ADMIN = `${ApiPaths.ADMIN_LEVEL}/enrollment`;
  public static readonly DASHBOARD_ADMIN = `${ApiPaths.ADMIN_LEVEL}/dashboard`;
  public static readonly EXAM_ADMIN = `${ApiPaths.ADMIN_LEVEL}/exam`;

  //teacher endpoints
  public static readonly COURSE_SECTION_TEACHER = `${ApiPaths.TEACHER_LEVEL}/course-section`;

  //student endpoint
  public static readonly ENROLLMENT_STUDENT = `${ApiPaths.STUDENT_LEVEL}/enrollment`;

  
  public static getFullUrl(path: string): string {
    const cleanPath = path.startsWith('/') ? path.slice(1) : path;
    return `${this.HOST}/${cleanPath}`;
  }
}
